File: State-of-artPizza/State-of-artPizza/PizzaStore.cpp
#include "PizzaStore.h"

unique_ptr<Pizza> PizzaStore::orderPizza(string type)
{
	unique_ptr<Pizza> pizza{ move(createPizza(type)) };

	pizza->prepare();
	pizza->bake();
	pizza->cut();
	pizza->box();

	return pizza;
}

File: PizzaWorld/src/PizzaStore/PizzaStore.java
package PizzaStore;

import Pizza.Pizza;
// import Factories.SimplePizzaFactory;

// public class PizzaStore {
//     SimplePizzaFactory factory;

//     public PizzaStore(SimplePizzaFactory simplePizzaFactory) {
//         factory = simplePizzaFactory;
//     }

//     public Pizza orderPizza(String type){
//         Pizza pizza = factory.createPizza(type);
        
//         pizza.prepare();
//         pizza.bake();
//         pizza.cut();
//         pizza.box();

//         return pizza;
//     }
// }


public abstract class PizzaStore {
    public Pizza orderPizza(String type){
        Pizza pizza = createPizza(type);

        pizza.prepare();
        pizza.bake();
        pizza.cut();
        pizza.box();

        return pizza;
    }

    abstract Pizza createPizza(String type);
}